tests/unit/Collectors/FrameworkCollectorTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+use AG\ElasticApmLaravel\Collectors\FrameworkCollector;
+use AG\ElasticApmLaravel\Collectors\RequestStartTime;
+use Codeception\Test\Unit;
+use Illuminate\Config\Repository as Config;
+use Illuminate\Foundation\Application;
+
+class FrameworkCollectorTest extends Unit
+{
+    /** @var Application */
+    private $app;
+
+    /** @var FrameworkCollector */
+    private $collector;
+
+    public function _before(): void
+    {
+        $this->app = app(Application::class);
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->collector = new FrameworkCollector(
+            $this->app,
+            new Config([]),
+            new RequestStartTime(0.0)
+        );
+    }
+
+    public function testCollectorName(): void
+    {
+        self::assertEquals('framework-collector', $this->collector->getName());
+    }
+
+    public function testItCanRegisterBootingEvent(): void
+    {
+        $this->markTestSkipped('Depending on the order tests are executed, Laravel boot event ');
+
+        $this->app->boot();
```

Comment:
Not sure how (I'm assuming `app(Application::class)` follows the singleton pattern), but depending on the order the tests are executed, the application is not [firing the callbacks](https://github.com/laravel/framework/blob/5.5/src/Illuminate/Foundation/Application.php#L765).

Cannot we make sure `$this->app` is always a new fresh instance?

Reply:
I've changed it to create a new instance, but I cannot really explain why it was failing before. [Also removed setUp as  it's not a codeception method](https://stackoverflow.com/questions/41934684/what-is-the-difference-between-setup-in-phpunit-and-before-in-codeception), but a phpunit one, so to remove any confusion about when that is run.

Reply:
> I've changed it to create a new instance, but I cannot really explain why it was failing before.
Nice, thanks!

>[Also removed setUp as it's not a codeception method](https://stackoverflow.com/questions/41934684/what-is-the-difference-between-setup-in-phpunit-and-before-in-codeception), but a phpunit one, so to remove any confusion about when that is run.
I'm going to update the rest of tests that are using `setup/teardown` as well.